pointcept/engines/test.py

```diff
@@ -883,3 +887,433 @@ def test(self):
     def collate_fn(batch):
         return collate_fn(batch)
+
+
+@TESTERS.register_module()
+class InstanceSegTest(TesterBase):
+    def __init__(self, cfg, **kwargs):
+        super().__init__(cfg, **kwargs)
+        self.segment_ignore_index = cfg.segment_ignore_index
+        self.sseg_class_names = cfg.class_names
+
+    def test(self):
+        assert self.test_loader.batch_size == 1
+        logger = get_root_logger()
+        logger.info(">>>>>>>>>>>>>>>> Start Evaluation >>>>>>>>>>>>>>>>")
+
+        evaluator = InsSegEvaluator(
+            segment_ignore_index=self.segment_ignore_index,
+            class_names=self.sseg_class_names,
+        )
+
+        self.model.eval()
+
+        save_path = os.path.join(self.cfg.save_path, "result")
+        make_dirs(os.path.join(save_path, "submit"))
+        for idx, data_dict in enumerate(self.test_loader):
+            input_dict = data_dict
+            print("Evaluating scene: ", f" [{idx+1}/50]")
+
+            self.model.eval()
+            with torch.no_grad():
+                self.move_batch_to_device(input_dict)
+                model_output = self.model(input_dict)
+                all_ap, all_ap_50, all_ap_25, loss = evaluator.eval(
+                    model_output, input_dict
+                )
+
+                print("All AP: ", all_ap)
+                print("All AP 50: ", all_ap_50)
+                print("All AP 25: ", all_ap_25)
+                print("Loss:", loss)
+                self.write_results(model_output, input_dict)
+
+    def move_batch_to_device(self, input_dict):
+        for key in input_dict.keys():
+            if isinstance(input_dict[key], torch.Tensor):
+                input_dict[key] = input_dict[key].cuda(non_blocking=True)
+
+    def write_results(self, model_output, input_dict):
+        scores = model_output["pred_scores"]
+        masks = model_output["pred_masks"]
+        classes = model_output["pred_classes"]
+        scan_name = input_dict["name"]
+        mask_dir = "predicted_masks"
+        scores[scores < 0] = 0
+        scores[scores > 1] = 1
+        # import pdb
+        # pdb.set_trace()
+        root_dir = os.path.join(self.cfg.save_path, "result", "submit")
+        mask_dir = os.path.join(root_dir, mask_dir)
+        make_dirs(mask_dir)
+        result_file_path = os.path.join(root_dir, f"{scan_name}.txt")
+
+        def to_numpy(tensor):
+            return tensor.cpu().numpy() if isinstance(tensor, torch.Tensor) else tensor
+
+        with open(result_file_path, "w") as result_file:
+            for i, (score, mask, cls) in enumerate(
+                zip(to_numpy(scores), to_numpy(masks), to_numpy(classes))
+            ):
+                binary_mask = mask.astype(np.uint8)
+                rle = self.rle_encode(binary_mask)
+
+                mask_filename = f"{scan_name}_{i:03d}.json"
+                mask_file_path = os.path.join(mask_dir, mask_filename)
+                with open(mask_file_path, "w") as mask_file:
```

Comment:
Hey @Streakfull 
This part of the code is a bit strange to me:

For each loop,  the code reopens the mask file and overwrite mask from previous loop. Does my understanding correct?